### PR TITLE
Master script speed up

### PR DESCRIPTION
**Pull request by MattCloward**
Here is a summary of some of the changes in this pull request:
- massive improvement in speed for creating the associations and study tables (6+ hrs to 1+ hr and 1+ hr to 10+ min respectively)
- updateDatabaseCron.sh script that calls the master script with parameters and is run by cron job on the first of every month
- master script now pushes to GitHub automatically near the end of the script finishing
- master script now has options for running with some of its functions disabled
- encoding of tables is now UTF-8

There are a lot of commits at the end because I was testing automatic git pushing.

### CODE REVIEW

File: update_database_scripts/README_master_script.txt [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
     d. The "8" in the command is the number of times the GWAS catalog will be split to download different parts of the database concurrently. 
        Increase this number to speed up the download, but don't increase the number too much since the website computer only has 2 cores.
-5. This command will take at least 4 hours to run. It is done when the last line of "output.txt" says "Press [Enter] key to finish..." At this point, 
-   all of the new data is already uploaded to the database and and you just need to upload the new TSVs to Github.
-6. Upload the new TSVs to github: (This may be another script in the future)
-    a. git add .
-    b. git commit -m "updated database *MM/DD/YY*"
-    c. git push origin master
-7. If you run the master script, but then want to stop it, you can use the following process:
+5. This command will take between 3-5 hours to run. It is done when the last line of "output.txt" says "Press [Enter] key to finish..." At this point,
```
Comment (mpage21) [inline update_database_scripts/README_master_script.txt:31]: Is this still between 3-5 hours?

File: update_database_scripts/createStudyTable.R [hunk 2]
```diff
@@ -1,262 +1,258 @@
-# This script uses data from an existing folder populated with CSV trait data and the GWAS catalog acccessable via the gwasrapidd library
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R:1]: Is this mostly the same except for the 'cat(..)' line at the bottom?

File: update_database_scripts/gitPush.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/expect -f
+
+# This script uses "expect" instead of bin to push committed files to the GitHub project. "except" allows the script to automatically return the GitHub
```
Comment (mpage21) [inline update_database_scripts/gitPush.sh:3]: instead of bin or instead of bash?

File: update_database_scripts/unpackDatabaseCommandLine.R [hunk 4]
```diff
@@ -1,459 +1,488 @@
...
-} else {
-  is_ebi_reachable(chatty = TRUE)
-  stop("The EBI API is unreachable. Check internet connection and try again.", call.=FALSE)
-}
-
-DevPrint(paste("Traits with no valid snps:", length(invalidStudies)))
-DevPrint(invalidStudies)
-print(paste0("Data download and unpack complete! ", "took: ", format(Sys.time() - start_time)))
\ No newline at end of file
+#!/usr/bin/env Rscript
```
Comment (mpage21) [inline update_database_scripts/unpackDatabaseCommandLine.R:1]: What lines are different in this file?

File: update_database_scripts/gitPush.sh (hunk 3 above)
Comment (MattCloward) [reply update_database_scripts/gitPush.sh]: Whoops! Bash. Fixed

File: update_database_scripts/README_master_script.txt (hunk 1 above)
Comment (MattCloward) [reply update_database_scripts/README_master_script.txt]: Yes, but it's the whole master script, not just the associations table. The associations table alone used to take 6+ hrs. Made a clarification in the documentation.

File: update_database_scripts/createStudyTable.R (hunk 2 above)
Comment (MattCloward) [reply update_database_scripts/createStudyTable.R]: This was one of the two most changed files. I fixed it so you can see the changes now.

File: update_database_scripts/unpackDatabaseCommandLine.R (hunk 4 above)
Comment (MattCloward) [reply update_database_scripts/unpackDatabaseCommandLine.R]: This was one of the two most changed files. I fixed it so you can see the changes now.

**Review by mpage21 — CHANGES_REQUESTED** [review]
See my comment about the typo. And then make sure you resolve the merge conflicts, and I think it is good to go after that! I love all the optional arguments for running the master script!

File: update_database_scripts/README_master_script.txt [hunk 5]
```diff
@@ -34,11 +42,17 @@ To update the PRSKB database automatically with the latest studies from the NHGR
 The following is a simplified description of the files in the update_database_scripts folder. They are listed in the order they are used when running 
 the master_script. Please see the beginning of each of the files for command line parameters, usage, and additional information.
 
+updaateDatabseCron.sh- A shell script that is run as a cron job the first day of each month. The cron job passes a password file path to the script
```
[the remark this answers is not in the document]
Comment (MattCloward) [reply update_database_scripts/README_master_script.txt]: fixed

**Pull request merged**